backend/src/test/java/jonaszeihe/ninjin/service/ElementServiceTest.java

```diff
@@ -0,0 +1,88 @@
+package jonaszeihe.ninjin.service;
+
+import jonaszeihe.ninjin.db.ElementMongoDb;
+import jonaszeihe.ninjin.model.Element;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.mockito.Mockito.*;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+class ElementServiceTest {
+
+    private final ElementMongoDb elementMongoDb = mock(ElementMongoDb.class);
+    private final ElementService elementService = new ElementService(elementMongoDb);
+
+    @Test
+    @DisplayName("Add new elements to the database")
+    public void testAddElements() {
+        //GIVEN
+        String newCourseName = "Yoga for beginners";
+        String newSegmentName = "Yoga for beginners 1";
+        String newElementContent = "some content";
+
+        //WHEN
+        elementService.addElements(newCourseName, newElementContent, newSegmentName);
+        //THEN
+        verify(elementMongoDb).save(Element.builder()
+                .elementName("Yoga for beginners 1 Element 1")
+                .elementContent("some content")
+                .segmentName("Yoga for beginners 1")
+                .build());
+        verify(elementMongoDb).save(Element.builder()
+                .elementName("Yoga for beginners 1 Element 2")
+                .elementContent("some content")
+                .segmentName("Yoga for beginners 1")
+                .build());
+        verify(elementMongoDb).save(Element.builder()
+                .elementName("Yoga for beginners 1 Element 3")
+                .elementContent("some content")
+                .segmentName("Yoga for beginners 1")
+                .build());
```

Comment:
This is more a general tip but you can see it quite clearly here: 
Try to extract those reoccuring builders to external methods to make the tests more readable.
For example you could create a factory class for this:
```java
public class TestElementFactory {

public static Element createYogaTestElement(String segment, String element) {
return Element.builder()
                .elementName("Yoga for beginners " + segment + " Element " + element)
                .elementContent("some content")
                .segmentName("Yoga for beginners " + segment)
                .build()
}
}
```

And then you could write your tests like this:
```java
verify(elementMongoDB).save(TestElementFactory.createYogaTestElement("1","1"));
verify(elementMongoDB).save(TestElementFactory.createYogaTestElement("1","2"));
...
```